library/src/ui/index.js

```diff
@@ -1 +1,72 @@
-window.CoNDeT.ui = {}
\ No newline at end of file
+window.CoNDeT.ui = {
+    BaseComponent: {
+        state: {
+
+        },
+        common: {
+
+        },
+        init: function(common, props) {
+            this.ref = this.onInit();
+        },
+        update: function(common, props) {
+            children.update();
+            this.onUpdate();
+            this.ref.onUpdate();
+        },
+        destroy: function(common, props) {
+            children.destroy();
+            this.onDestroy();
+            this.ref.onDestroy();
+        },
```

Comment:
Po namyśle, w `destroy` będzie podawany jedynie `common` bez `props`. Poprawię to w dokumentacji  
Przed `children.update();` nie ma `this` więc tworzysz zmienną globalną. Dodatkowo children jest tablicą, więc w praktyce należy:  
1. sprawdzić, czy `this.children` istnieje (minimum, że jest `!= null` co sprawdzi jednocześnie, czy nie jest `undefined`, można sprawdzić też, czy jest tablicą, ale nie wiem, czy jest taka potrzeba, chyba nie)
2. iteracja po elementach tablicy
3. dla każdego wywołanie `update` z przekazanymi common  
`this.onDestroy();` podać `common` a `this.ref.onDestroy();` usunąć (tłumaczyłem dlaczego komentarz wyżej)